packages/evolution-generator/src/scripts/conditionals_generator.py

```
@@ -270,6 +285,16 @@ def _collect_row_validation_issues(
                         f"must be one of {sorted(spec.allowed_values - {None})!r} or empty, got {repr(cell_value)}"
                     )
 
+        parentheses_value = self._empty_to_none(row_dict.get("parentheses"))
+        if isinstance(parentheses_value, str) and not self._is_valid_parentheses_cell(
+            parentheses_value
+        ):
+            issues.append(
+                f"{prefix}Invalid parentheses in row {row_number}: "
+                "must contain only '(' and ')' characters with openings before closings, "
+                f"or be empty, got {repr(parentheses_value)}"
```

Ruff (0.16.0)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)